frontend-web/webclient/app/Applications/Jobs/View.tsx

```diff
@@ -1399,37 +1487,48 @@ const RunningButtonGroup: React.FunctionComponent<{
                 </Button>
             </Link>
         )}
-        {appType !== "WEB" || !supportsInterface ? null : (
-            <Link to={redirectToWeb ?? ""} aria-disabled={!redirectToWeb} target={"_blank"}>
-                <Button width="220px" disabled={!redirectToWeb}>
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
-        )}
-        {appType !== "VNC" || !supportsInterface ? null : (
-            <Link to={`/applications/vnc/${job.id}/${rank}?hide-frame`} target={"_blank"} onClick={e => {
-                e.preventDefault();
-
-                const link = findDomAttributeFromAncestors(e.target, "href");
-                if (!link) return;
-
-                window.open(
-                    link,
-                    `vnc-${job.id}-${rank}`,
-                    "width=800,height=450,status=no"
-                );
-            }}>
-                <Button width="220px">
-                    <Icon name={"heroArrowTopRightOnSquare"} />
-                    <div>Open interface {hasMultipleNodes ? `(node ${rank + 1})` : null}</div>
-                </Button>
-            </Link>
+        {interfaceLinks.length < 1 ? <></> : (
+            <Flex>
+                <Link to={interfaceLinks[0].link ?? ""} aria-disabled={!interfaceLinks[0]} target={"_blank"}>
+                    <Button attachedLeft={interfaceLinks.length > 1} disabled={!interfaceLinks[0]}>
+                        <Icon name="heroArrowTopRightOnSquare" />
+                        <div style={{minWidth: interfaceLinks.length > 1 ? "130px" : "164px"}}>{interfaceLinks[0].target ?? ("Open interface" + (hasMultipleNodes ? ` (node ${rank + 1})` : null))}</div>
+                    </Button>
+                </Link>
+                <ClickableDropdown
+                    trigger={
+                        <div className={InterfaceSelectorTrigger}> 
+                            <Icon name="chevronDownLight"/>
+                        </div>
+                    }
+                    right={"0px"}
+                    minWidth={"245px"}
+                    paddingControlledByContent
+                >
+                    {interfaceLinks.slice(1).map((link, k) => (
```

Comment:
I would filter out empty links. A link to a non-existant location seems like a weird user experience.

`interfaceLinks.slice(1).filter(it => it.link).map((link, k) => (`

Reply:
The job of `findInterfaceTargets` is to populate the `link` property and should never be undefined at this point.